### PR TITLE
RAC Select: propagate placeholder to Button label

### LINKED ISSUE #4398: RAC - Select and SelectValue do not honor `placeholder` prop
# 🐛 Bug Report 

For the `react-aria-components` `Select` and `SelectValue` components, setting the `placeholder` prop on either one does not affect the rendered HTML. Looking at the source code, it seems that the placeholder is hard-coded to `Select an item`.

Secondarily, it is a bit confusing as to which component the placeholder should be set on. It seems that the intent is that the `Select` component receives the `placeholder` prop rather than `SelectValue`.

## 🤔 Expected Behavior

Setting the `placeholder` prop on `Select` should display a placeholder when no item is selected.

The `SelectValue` component should not accept a `placeholder` prop.

## 😯 Current Behavior

Neither the `Select` nor `SelectValue` `placeholder` prop have any effect.

## 💁 Possible Solution

Change the `defaultChildren` render prop in `SelectValue` to accept the `placeholder` prop, if defined. It's possible that this needs to somehow come from the `SelectContext` if the intent is for the prop to be set on the `Select` component.

https://github.com/adobe/react-spectrum/blob/b92f6029e71aa29a9f7b35e403586856a90004ad/packages/react-aria-components/src/Select.tsx#L154

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-aria-components  | 1.0.0-alpha.3 |
| Browser          | Chrome 112.0.5615.49 (Official Build) (arm64) |
| Operating System | macOS 13.3.1 |

### PR DESCRIPTION
**Pull request by reidbarber**
Closes #4398

If no item is selected and a placeholder is provided, the button should use that as it's label instead of the default.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/react-aria-components/src/Select.tsx [hunk 1]
```diff
@@ -151,7 +152,7 @@ function SelectValue<T extends object>(props: SelectValueProps<T>, ref: Forwarde
   let renderProps = useRenderProps({
     ...props,
     // TODO: localize this.
-    defaultChildren: rendered || 'Select an item',
+    defaultChildren: rendered || placeholder || 'Select an item',
```
Comment (ktabors) [inline packages/react-aria-components/src/Select.tsx:155]: I thought `Select an item` was going to be a translation, but it looks like we aren't doing default translations.

**Review by ktabors — APPROVED** [review]
LGTM. I wish the storybook for RAC select had controls for us to set the various props for better testing.

**Comment by reidbarber** [comment]
> I wish the storybook for RAC select had controls for us to set the various props for better testing.

@ktabors agreed, maybe we can convert this file to CSF3 soon

**Review by LFDanLu — APPROVED** [review]
LGTM

**Pull request merged**